### PR TITLE
Conditional Notebook Test Trigger

### PR DESCRIPTION
**Pull request by wd60622**
Similar to #1153

Are these the right conditions?

----
📚 Documentation preview 📚: https://pymc-marketing--1163.org.readthedocs.build/en/1163/

### CODE REVIEW

**Comment by wd60622** [comment]
Sorry @junpenglao , typed out Ju...an

File: .github/workflows/test_notebook.yml [hunk 1]
```diff
@@ -4,13 +4,15 @@ on:
   pull_request:
     branches: [main]
     paths:
-      - '**.py'
-      - '**.ipynb'
+     - "pyproject.toml"
+     - "tests/**.py"
+     - "pymc_marketing/**"
```
Comment (juanitorduz) [inline .github/workflows/test_notebook.yml:9]: But we also need to be triggered if the notebooks from the docs change themselves, right?

Comment (wd60622) [reply .github/workflows/test_notebook.yml:9]: Good point. Should I include changes to just tests?

- \"pyproject.toml\" # Environment changes (the ones used in CI)
- \"pymc_marketing/**\" # Changes to module
- \"docs/**.ipynb\" # Changes to notebooks in the documentation

Comment (wd60622) [reply .github/workflows/test_notebook.yml:9]: Good point. Should I include changes to just tests?

- \"pyproject.toml\" # Environment changes (the ones used in CI)
- \"pymc_marketing/\\*\\*\" # Changes to module
- \"docs/**.ipynb\" # Changes to notebooks in the documentation

Comment (wd60622) [reply .github/workflows/test_notebook.yml:9]: Good point. Should I include changes to just tests?

- \"pyproject.toml\" # Environment changes (the ones used in CI)
- \"pymc_marketing/\\*\\*\" # Changes to module
- \"docs/\\*\\*.ipynb\" # Changes to notebooks in the documentation

Comment (juanitorduz) [reply .github/workflows/test_notebook.yml:9]: maybe add explicitly `"docs/source/notebooks/mmm` (and clv and others). The reason is that we want to omit the dev notebooks https://github.com/pymc-labs/pymc-marketing/tree/main/docs/source/notebooks/clv/dev . Or any other way we make sure we omit just these dev ones

File: .github/workflows/test_notebook.yml [hunk 2]
```diff
@@ -4,13 +4,19 @@ on:
...
   push:
     branches: [main]
     paths:
-      - '**.py'
-      - '**.ipynb'
+     - "pyproject.toml"
+     - "tests/**.py"
+     - "pymc_marketing/**"
+     - "docs/source/notebooks/**.ipynb"
+     - "!docs/source/notebooks/*/dev/**.ipynb"
```
Comment (juanitorduz) [inline .github/workflows/test_notebook.yml:19]: I did not know this negation was possible :)

**Review by juanitorduz — APPROVED** [review]

File: .github/workflows/test_notebook.yml (hunk 2 above)
Comment (wd60622) [reply .github/workflows/test_notebook.yml:19]: In the docs [here](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#example-including-and-excluding-paths)

**Pull request merged**